TOT-101 : [FIX] Step1 에러 디버깅 및 추가사항 구현

yu-ratel opened the pull request:
## PR전 체크리스트
> 1. PR 및 Commit title을 형식에 맞게 작성했나요(e.g. TOT-123 : [TYPE])
✅
> 2. Reviewers가 명확하게 지정됐나요?
✅
> 3. Assignees가 명확하게 지정됐나요?
✅


## 📝 작업 내용

- [x] 날짜 선택 안해도 일정 만들기 버튼 클릭가능
- [x] 날짜 선택시 주말 색상 미적용 
- [x] step2, 3에서 step1으로 돌아갈 시 선택했던 데이터를 날리고 날짜를 다시 만들것인지 경고창 필요해보임 (+ 데이터가 날아가야함) 
- [x] 날짜만들기 날짜선택시 현재 달이 아니면 현재 달로 가짐 
- [x] 날짜 선택하지 않는 경우 2월로 나타남

+ 추가 에러사항 및 수정사항 
- [x] 선택한 요일이 달을 바꿔도 그 달의 요일로 렌더되는 현상 지우기
- [x] 경고 모달 하나로 통일 (예, 아니오 방식)

### 스크린샷

## 💬 리뷰 요구사항

## 참고자료

yu-ratel committed:
TOT-101 : [FIX] 날짜 선택하지않을시 2월로 보이는 오류 디버깅
totalPlanAtom의 기본값을 2월에서 빈배열로 설정

yu-ratel committed:
TOT-101 : [FIX] 날짜 선택 하지않아도 날짜 설정하기버튼이 활성화되어있는 오류 디버깅
비활성화 버튼을 만들기 위한 공통버튼 스타일 export 추가
createScheduleAtom을 이용하여 활성화 비활성화 판단가능하게 추가

yu-ratel committed:
TOT-101 : [FIX] 날짜 선택시 현재달로 강제 렌더링 되는 현상, 주말 색상 미적용 디버깅
기본 최대날짜 설정 및 최대날짜보다 작은경우에만 색상지정

yu-ratel committed:
TOT-101 : [FIX] 현재 날짜, 선택된 날짜 해당월이 아닌 다음달에도 표시되는 에러 디버깅
선택된 날짜 css unset으로 돌린 뒤 css우선순위를 위한 위치변경

yu-ratel committed:
TOT-101 : [REFACTOR] 재사용을 위한 LoginModal props 변경

yu-ratel committed:
TOT-101 : [FEAT] step2, 3에서 step1 선택시 경고모달 구현

yu-ratel committed:
TOT-101 : [REFACTOR] 통일성을 목적으로 경고모달 하나로 통일
LoginModal 삭제

yu-ratel commented on frontend/src/components/StepNavBar/StepNavBar.tsx:
```diff
@@ -12,15 +14,29 @@ const StepNavBar = () => {
-            <LinkItem>{`step${idx + 1}`}</LinkItem>
-            <LinkItem>{item.content}</LinkItem>
-          </LinkBox>
+          {idx === 0 ? (
+            <ExtendBox $isClicked={idx + 1 === currentStep} onClick={onClick}>
+              <LinkItem>{`step${idx + 1}`}</LinkItem>
+              <LinkItem>{item.content}</LinkItem>
+            </ExtendBox>
+          ) : (
+            <ExtendLinkBox to={`${item.path}/${regionId}`} $isClicked={idx + 1 === currentStep}>
+              <LinkItem>{`step${idx + 1}`}</LinkItem>
+              <LinkItem>{item.content}</LinkItem>
+            </ExtendLinkBox>
+          )}
```
이 부분이 반복문으로 만들어진 컴포넌트라서 외부에서는 접근이 어려웠고, 조건문에 의해 step1클릭이 가능하게 했을 때 Link버튼이라 Step1을 가서 경고모달을 띄워주는식으로 작업해야 했기에  공용 style을 만들고 step1만 버튼으로 만들어준다음 동작하게 했습니다.

yu-ratel commented on frontend/src/components/StepNavBar/StepNavBar.tsx:
```diff
@@ -89,3 +105,11 @@ const LinkBox = styled(Link)<{ $isClicked: boolean }>`
+
+const ExtendBox = styled.div<{ $isClicked: boolean }>`
+  ${stepInner}
+`;
```
만들어주신 컴포넌트를 css로 지정하여 상속가능하게 한 뒤 사용했습니다.

rlaclghks123 replied:
여기에 cursor: pointer 속성이 하나 추가되면 좋을 것 같습니다!

kyejin0412 replied:
배우고 갑니다!!

yu-ratel reviewed (commented):
step 건든 부분입니다!

rlaclghks123 commented on frontend/src/components/AlertModal/AlertModal.tsx:
```diff
@@ -1,35 +1,71 @@
-          <LinkButton path="/" color={theme.colors.darkGrey}>
-            홈으로
+          <LinkButton path={path} color={theme.colors.mainColor}>
+            {onButtonText}
```
이렇게 재활용 하는군요! 배워갑니다!

yu-ratel committed:
Merge branch 'develop' into FIX/TOT-101

yu-ratel committed:
TOT-101 : [FIX] git 충돌해결
develop pull 후 수정

yu-ratel committed:
TOT-101 : [FIX] git 충돌사항 해결
완벽히 바꾸지 않아 stpe이 메인컬러로 변하지않는 현상 해결

yu-ratel committed:
TOT-101 : [FIX] 화면에서 로고클릭시 스크롤 안되는 현상 해결
로고버튼 클릭시 전역으로 주는 모달상태변화값을 메인에선 적용되지 않게하고 메인에서 클릭시 최상단으로 스크롤 되게 적용
step 버튼을 제외한 404 에러 페이지등 모든 페이지에서 기본값 홈으로 돌아가기 모달 적용

yu-ratel committed:
TOT-101 : [REFACTOR] 무한스크롤 훅 수정
initialPageParam: 0 -> initialPageParam: 1

rlaclghks123 reviewed (approved):
확인했습니다! 고생하셨어요!

kyejin0412 reviewed (approved):
고생하셨습니다!!

yu-ratel committed:
Merge remote-tracking branch 'origin' into FIX/TOT-101

yu-ratel committed:
merge
